source/extensions/transport_sockets/tls/cert_validator/spiffe/spiffe_validator.cc

```diff
@@ -30,10 +33,128 @@ namespace Tls {
 
 using SPIFFEConfig = envoy::extensions::transport_sockets::tls::v3::SPIFFECertValidatorConfig;
 
+std::shared_ptr<SpiffeData>
+SPIFFEValidator::parseTrustBundles(const std::string& trust_bundle_mapping_str) {
+  Json::ObjectSharedPtr parsed_json_bundle;
+
+  ENVOY_LOG(info, "Parsing trust_bundles");
+
+  auto json_parse_result = Envoy::Json::Factory::loadFromString(trust_bundle_mapping_str);
+  if (!json_parse_result.ok()) {
+    ENVOY_LOG(error, "Failed to parse SPIFFE bundle map JSON");
+    return nullptr;
+  }
+
+  parsed_json_bundle = json_parse_result.value();
+
+  std::shared_ptr<SpiffeData> spiffe_data = std::make_shared<SpiffeData>();
+
+  const auto trust_domains = parsed_json_bundle->getObject("trust_domains");
+
+  if (!trust_domains.ok() || *trust_domains == nullptr || (*trust_domains)->empty()) {
+    ENVOY_LOG(error, "No trust domains found in SPIFFE bundle map");
+    return nullptr;
+  }
+
+  bool success = true;
```

Comment:
rather than simple bool, please use a absl::Status here to record the error of the parsing.